lib/klaviyo/client.rb

```diff
@@ -31,44 +192,144 @@ def track(event, kwargs = {})
         :ip => ''
       }
       params[:time] = kwargs[:time].to_time.to_i if kwargs[:time]
-     
-      params = build_params(params)
-      request('api/track', params)
+
+      request('GET', 'track', params)
     end
-    
+
     def track_once(event, opts = {})
       opts.update('__track_once__' => true)
       track(event, opts)
     end
-    
+
     def identify(kwargs = {})
       defaults = {:id => nil, :email => nil, :properties => {}}
       kwargs = defaults.merge(kwargs)
-      
+
       if kwargs[:email].to_s.empty? and kwargs[:id].to_s.empty?
         raise KlaviyoError.new('You must identify a user by email or ID')
       end
-      
+
       properties = kwargs[:properties]
       properties[:email] = kwargs[:email] unless kwargs[:email].to_s.empty?
       properties[:id] = kwargs[:id] unless kwargs[:id].to_s.empty?
 
-      params = build_params({
+      params = {
         :token => @api_key,
         :properties => properties
-      })
-      request('api/identify', params)
+      }
+
+      request('GET', 'identify', params)
     end
 
     private
-    
+
+    def request(method, path, kwargs = {})
```

Comment:
overalll thought here, i'm trying to find a way if we could simplify the actual request method by doing a faraday connection rather than a bunch of elsif statements to determine the method..

Does it make sense that v2_request. v1_request and then create a new public_request (for the track/identify) calls that build send in a path w/ a method string.

so something like:
```
def request(method, path, params = {}, body = {})
    con = Faraday.new(
        url: URL_HERE.
        params: params,
        headers: {# user agent, maybe content type?}
    )
    response = con.send(method.downcase) do |req| # i didn't test this as an fyi
      req.body = body.to_json
    end
```
the hope of the above is that we make the code smart enough and don't care about what HTTP method get's passed through, it just handles it!